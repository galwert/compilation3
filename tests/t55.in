// this is ok :)
int getInput() {
    return 1; // TODO - Actually read input from user
}

void main(){
    int x = getInput();
    int y = getInput();
    int XplusY = x + y;
    int XdivY = x / y;
    bool XeqY = x == y;
    bool XgtY = x > y;
    bool XltY = x < y;
    bool xneY = x != y;

    if (x != y) {
        return;
    }

    x = getInput();
    y = getInput();

    XgtY = x >= y;
    XltY = x <= y;

}