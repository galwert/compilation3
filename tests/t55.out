---end scope---
---end scope---
---end scope---
---end scope---
x INT 0
y INT 1
XplusY INT 2
XdivY INT 3
XeqY BOOL 4
XgtY BOOL 5
XltY BOOL 6
xneY BOOL 7
---end scope---
print (STRING)->VOID 0
printi (INT)->VOID 0
getInput ()->INT 0
main ()->VOID 0
